File: M8/Exemplos/App_eXperience_Java/app/src/main/java/com/elgin/elginexperience/MainActivity.java
package com.elgin.elginexperience;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.zxing.oned.CodaBarReader;

public class MainActivity extends AppCompatActivity {
    static Context context;

    Button buttonPrinterOption;
    Button buttonTefOption;
    Button buttonBarCodeReaderOption;
    Button buttonSAT;
    Button buttonShipay;
    Button buttonBalancaOption;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        context = this;

        buttonPrinterOption = findViewById(R.id.buttonPrinterOption);
        buttonTefOption = findViewById(R.id.buttonTefOption);
        buttonBarCodeReaderOption = findViewById(R.id.buttonBarCodeReaderOption);
        buttonSAT = findViewById(R.id.buttonSAT);
        buttonShipay = findViewById(R.id.buttonShipay);
        buttonBalancaOption = findViewById(R.id.buttonBalancaOption);

        buttonPrinterOption.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { openPrinterScreen(); }
        });

        buttonTefOption.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { openTefScreen(); }
        });

        buttonBarCodeReaderOption.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { openBarCodeReaderScreen(); }
        });

        buttonSAT.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { openSATScreen(); }
        });

        buttonShipay.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { openShipayScreen(); }
        });

        buttonBalancaOption.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { openBalancaScreen(); }
        });
    }

    public void openPrinterScreen() {
        Intent intent = new Intent(this, PrinterMenu.class);
        startActivity(intent);
    }

    public void openTefScreen() {
        Intent intent = new Intent(this, Tef.class);
        startActivity(intent);
    }

    public void openBarCodeReaderScreen(){
        Intent intent = new Intent(this, BarCodeReader.class);
        startActivity(intent);
    }

    public void openSATScreen(){
        Intent intent = new Intent(this, SatPage.class);
        startActivity(intent);
    }

    public void openShipayScreen() {
        Intent intent = new Intent(this, ShipayMenu.class);
            startActivity(intent);
    }

    public void openBalancaScreen(){
        Intent intent = new Intent(this, BalancaPage.class);
        startActivity(intent);
    }
}

File: M8/Exemplos/App_eXperience_Kotlin/app/src/main/java/com/example/e1_elgin_kotlin/MainActivity.kt
package com.example.e1_elgin_kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    lateinit var buttonPrinterOption: Button
    lateinit var buttonTefOption: Button
    lateinit var buttonBarCodeReaderOption: Button
    lateinit var buttonSAT: Button
    lateinit var buttonBalancaOption: Button
    lateinit var buttonShipay: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        buttonPrinterOption = findViewById(R.id.buttonPrinterOption)
        buttonTefOption = findViewById(R.id.buttonTefOption)
        buttonBarCodeReaderOption = findViewById(R.id.buttonBarCodeReaderOption)
        buttonSAT = findViewById(R.id.buttonSAT)
        buttonBalancaOption = findViewById(R.id.buttonBalancaOption);
        buttonShipay = findViewById(R.id.buttonShipay);


        buttonPrinterOption.setOnClickListener {
            val intent = Intent(this, PrinterMenu::class.java)
            startActivity(intent)
        }

        buttonTefOption.setOnClickListener {
            val intent = Intent(this, Tef::class.java)
            startActivity(intent)
        }

        buttonBalancaOption.setOnClickListener {
            val intent = Intent(this, BalancaPage::class.java)
            startActivity(intent)
        }

        buttonBarCodeReaderOption.setOnClickListener {
            val intent = Intent(this, BarCodeReader::class.java)
            startActivity(intent)
        }
        buttonSAT.setOnClickListener {
            val intent = Intent(this, SatPage::class.java)
            startActivity(intent)
        }
        buttonShipay.setOnClickListener {
            val intent = Intent(this, ShipayMenu::class.java)
            startActivity(intent)
        }
    }
}



